### PR TITLE
[OSJC-283] - Binary build support

### PR DESCRIPTION
**Pull request by jeffmaury**
Signed-off-by: Jeff MAURY <jmaury@redhat.com>

### CODE REVIEW

File: src/main/java/com/openshift/restclient/model/build/IBinaryBuildSource.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+
+package com.openshift.restclient.model.build;
+
+public interface IBinaryBuildSource extends IBuildSource {
+
+    /**
+     * The asFile
+     *
```
Comment (adietish) [inline src/main/java/com/openshift/restclient/model/build/IBinaryBuildSource.java]: I'd remove this javadoc as it offers no added value or, alternatively comment what this method does.

File: src/main/java/com/openshift/internal/restclient/model/Build.java [hunk 2]
```diff
@@ -91,7 +92,9 @@ public String getOutputKind() {
         switch (asString("spec.source.type")) {
         case BuildSourceType.GIT:
             return (T) new GitBuildSource(asString("spec.source.git.uri"), asString("spec.source.git.ref"),
-                    asString("spec.source.git.contextDir"));
+                    asString("spec.source.contextDir"));
```
[the remark this answers is not in the document]
Comment (jeffmaury) [reply src/main/java/com/openshift/internal/restclient/model/Build.java]: No the code you mention uses s.build.Spec.Source.ContextDir and not s.build.Spec.Source.Git.ContextDir

File: src/main/java/com/openshift/internal/restclient/DefaultClient.java [hunk 3]
```diff
@@ -267,6 +302,46 @@ public String getResourceURI(IResource resource) {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    public <T> T execute(ITypeFactory factory, String method, String kind, String version, String namespace, String name,
```
[the remark this answers is not in the document]
Comment (jeffmaury) [reply src/main/java/com/openshift/internal/restclient/DefaultClient.java]: OK, will do and thanks for the snippet

**Comment by adietish** [comment]
[merge]

**Pull request merged**